doc/TockBinaryFormat.md

```diff
@@ -364,10 +372,28 @@ included, so long as no `offset` is repeated for a single driver. When
 multiple `offset`s and `allowed_commands`s are used they are ORed together,
 so that they all apply.
 
+#### `7` Kernel Version
+
+The `Kernel Version` header is designed to prevent the kernel
+from running applications that are not compatible with it.
+
+It defins the following two items:
+* `Kernel major` or `V` is the kernel major number (for Tock 2.0, it is 2)
+* `Kernel minor` or `v` is the kernel minor number (for Tock 2.0, it is 0)
+
+Apps defining this header are compatible with kernel version ^V.v (>= V.v and < (V+1).0)
```

Comment:
I am wary of this kind of specification. For example, it means that we can't change a system call API for a particular device number without a major number increment. Defining "compatible" seems difficult. 

Instead, I'd suggest that we add this to the end of the main header. It should either be present in every app, or not. It should include only a major version number, which relates to the ABI with the kernel. Anything further than that (compatibility) can be determined by the app itself, e.g., by probing if devices are installed.

Reply:
The intent is for this (i.e. TLV id 7) to only specify kernel ABI. We wanted major and minor, because Tock 2.1 might include a new allow call that Tock 2.0 does not. We do need to make this clear in the docs.

We also thought that a TLV was an appropriate option for two reasons:
1. We're not entirely sure this is the long term best way to handle this sort of compatibility checking. Using a TLV makes it easy to deprecate this in the future and possibly replace it with something else.
2. Some boards/deployments might not want to include any compatibility information (and save the bytes in the header). They can easily omit the TLV.

Reply:
I detailed this in the documentation.